components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java

```diff
@@ -234,6 +234,7 @@ public static ServiceProvider build(OMElement serviceProviderOM) {
             }
         }
 
+        serviceProvider.setApplicationResourceId(serviceProvider.getApplicationName());
```

Comment:
shall we move this logic to https://github.com/wso2/carbon-identity-framework/blob/ab205d1a9df285288fd3e65d6b62fe39f2e0e9bf/components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java#L152-L153